pandas/_libs/tslibs/strptime.pyx

```diff
@@ -60,15 +64,27 @@ cdef bint format_is_iso(f: str):
     Generally of form YYYY-MM-DDTHH:MM:SS - date separator can be different
     but must be consistent.  Leading 0s in dates and times are optional.
     """
-    excluded_formats = ["%Y%m"]
-
-    for date_sep in [" ", "/", "\\", "-", ".", ""]:
-        for time_sep in [" ", "T"]:
-            for micro_or_tz in ["", "%z", ".%f", ".%f%z"]:
-                iso_fmt = f"%Y{date_sep}%m{date_sep}%d{time_sep}%H:%M:%S{micro_or_tz}"
-                if iso_fmt.startswith(f) and f not in excluded_formats:
-                    return True
-    return False
+    iso_regex = re.compile(
+        r"""
+        ^                     # start of string
+        %Y                    # Year
+        (?:([-/ \\.]?)%m      # month with or without separators
+        (?:\1%d               # day with or without separators
+        (?:[ |T]%H            # hour with separator
+        (?:\:%M               # minute with separator
+        (?:\:%S               # second with separator
+        (?:%z|.%f(?:%z|Z)?    # timezone or fractional second
+        )?)?)?)?)?)?          # optional
+        $                     # end of string
+        """,
+        re.VERBOSE,
+    )
+    excluded_formats = [
+        r"^%Y%m$",
+    ]
+    if any(re.match(pattern, f) for pattern in excluded_formats):
+        return False
+    return bool(re.match(iso_regex, f))
```

Comment:
I don't think we need a regex for the excluded formats - would
```python
return re.match(iso_regex, f) is not None and f not in excluded_formats
```
work?

Reply:
yea worked fine without the regex excluded_format